src/app/shared/services/volume.service.ts

```diff
@@ -132,6 +132,7 @@ export class VolumeService extends BaseBackendService<Volume> {
   }
 
   public markForRemoval(volume: Volume): Observable< Volume> {
+    volume.snapshots.forEach((snapshot) => this.snapshotService.markForRemoval(snapshot).subscribe());
```

Comment:
This should not work like this. We should make all the async requests. and subscribe to the observable